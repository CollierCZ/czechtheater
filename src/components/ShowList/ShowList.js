import React from "react"
import ShowCard from "../ShowCard/ShowCard"

class ShowList extends React.Component {
  render() {
    const { shows, future } = this.props;
    console.log(shows)
    return (
      shows.map(show => {
        console.log(show.elements.short_description)
        let propsArray = {
          name: show.elements.name.value,
          description: show.elements.short_description.value,
          imageUrl: show.elements.main_image.assets[0].url,
          imageDescription: show.elements.main_image.assets[0].description,
          slug: show.elements.url.value
        }
        future ? propsArray.time = "future" : propsArray.time = "past";
        show.elements.ticket_link.value ? propsArray.ticketLink = show.elements.ticket_link.value : propsArray.ticketLink = false;
        if (show.elements.gallery.assets) { propsArray.gallery = show.elements.gallery.assets };
        return ( <ShowCard key = { propsArray.name } { ...propsArray } />
        )
      }))
  }
}

export default ShowList