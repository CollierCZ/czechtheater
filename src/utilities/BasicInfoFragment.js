import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


