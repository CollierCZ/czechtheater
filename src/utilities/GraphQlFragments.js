import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


