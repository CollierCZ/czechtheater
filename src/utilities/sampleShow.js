export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }