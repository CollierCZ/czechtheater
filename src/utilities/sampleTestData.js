export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}