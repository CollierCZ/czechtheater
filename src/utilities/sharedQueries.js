import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`