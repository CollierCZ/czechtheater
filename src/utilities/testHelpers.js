import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}