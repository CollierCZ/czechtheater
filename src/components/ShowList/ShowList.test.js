import React from "react";
import { sampleShow } from "../../utilities/sampleShow";
import { shallow } from "enzyme";
import ShowList from "./ShowList";

const shows = [sampleShow];

describe("ShowList", () => {
  it("loads correctly for future shows", () => {
    const component = shallow (<ShowList shows={shows} future="true" />)
    expect(component).toMatchSnapshot()
  })
  it("loads correctly for past shows", () => {
    const component = shallow (<ShowList shows={shows} />)
    expect(component).toMatchSnapshot()
  })
});