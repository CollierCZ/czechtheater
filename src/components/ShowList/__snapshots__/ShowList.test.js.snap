// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowList loads correctly for future shows 1`] = `
<ShowCard
  description="All the acting"
  gallery={
    Array [
      Object {
        "description": "A cool image",
        "url": "https://example.com",
      },
    ]
  }
  imageDescription="A cool image"
  imageUrl="https://example.com"
  key="Alquist's revenge"
  name="Alquist's revenge"
  slug="alquists-revenge"
  ticketLink="https://example.com"
  time="future"
/>
`;

exports[`ShowList loads correctly for past shows 1`] = `
<ShowCard
  description="All the acting"
  gallery={
    Array [
      Object {
        "description": "A cool image",
        "url": "https://example.com",
      },
    ]
  }
  imageDescription="A cool image"
  imageUrl="https://example.com"
  key="Alquist's revenge"
  name="Alquist's revenge"
  slug="alquists-revenge"
  ticketLink="https://example.com"
  time="past"
/>
`;
